### PR TITLE
limit amount of commits displayed

### PR DESCRIPTION
**Pull request by sfahlberg**
### Changes
 * add global variable for the amount of commits displayed for git log
 * limit the number of commits by that amount

 ### Rationale
 * When I ran git log from merginal, it was taking a long time because of all the commits in the repo. This allows a sensible default (20 commits) and the option for the user to change the amount of commits displayed

### CODE REVIEW

**Comment by sfahlberg** [comment]
@idanarye any thoughts on this PR?

**Comment by idanarye** [comment]
Oh, yea, sorry for neglecting this. I wanted to write a long answer but there was always something more urgent to do and I forgot...

Anyways, I have some remarks about the code - which I will put on the diff. But more pressing are the conceptual remarks:

* 20 is not a sensible default. It may be good enough for single-developer hobby projects - but in my main repo at my workplace, for example, 20 commits are less than half a day.
    * That repo, BTW, has about 35k commits - and my laptop loads and displays them all in the Merginal buffer in one second.
    * But - when I tried a bigger project - like the Linux kernel, with 468k commits(my checkout is 3 years old - I don't actually contribute to it, I just use it as a huge project example when testing Vim plugins) - it took 10 seconds. So... you do have a case here.
* When limiting a list a UI should always inform the user there are more items and allow them to fetch the next batch.
* Merginal supports filtering(press `&` in any merginal buffer). This is done with Vim's regex, so it is always done **after** it receives the data. If you filter for "Alice" to see her commits, but she doesn't have any recent ones because she was working on a side branch for the last week - you will see nothing even though she had countless commits in the master branch.

Due to these concerns, I consider the current PR as **breaking** Merginal's history log future. But - since it does take too long(to you, at least), I suggest this:

1. You'll set the default to _unlimited_(any negative number if you pass it as argument to `--max-count`).
2. You(or I) will document the two caveats(no way to see next results, filtering done on the cropped list) in `doc/merginal.txt`.
3. When I'll have time, I'll implement a the missing features:
    * _SHOW MORE_ functionality.
    * Workaround for the crop-then-filter problem(fetch more results if we don't have enough?)
4. After that, I can set the default for 1024 - enough results to actually provide value while still something that can be fetched fast enough.

File: autoload/merginal/buffers/historyLog.vim [hunk 1]
```diff
@@ -12,8 +12,10 @@ function! s:f.generateHeader() dict abort
                 \ '']
 endfunction
 
+let g:merginal_log_commit_count = get(g:, 'merginal_log_commit_count', -20)
```
Comment (idanarye) [inline autoload/merginal/buffers/historyLog.vim]: * The setting should be named `g:merginal_logCommitCount` - this is the style I use in all my plugins.
* I don't like that set-a-default-somewhere global approach - it's not that hard to pick the default at the point of usage, and it allows the user to return to default with a simple `:unlet`.
* Using a negative number to feed it to the `-{number}` short style is just horrible. If a user wants to set this flag, the intuitive way would be `let g:merginal_logCommitCount = 100` - but this will break the history log and the user will not know why. Use the `--max-count` flag instead.

File: autoload/merginal/buffers/historyLog.vim [hunk 2]
```diff
@@ -12,8 +12,10 @@ function! s:f.generateHeader() dict abort
...
+
 function! s:f.generateBody() dict abort
-    let l:logLines = self.gitLines('log', '--format=%h %aN%n%ci%n%s%n', self.branch, '--')
+    let l:logLines = self.gitLines('log', g:merginal_log_commit_count,'--format=%h %aN%n%ai%n%s%n', self.branch)
```
Comment (idanarye) [inline autoload/merginal/buffers/historyLog.vim]: Why are you changing committer-time to author-time? I'm not going to debate which one is superior, but if you want such a change it'll need to be a different PR and it'll need to be configurable(because I prefer commiter-time)

Comment (idanarye) [inline autoload/merginal/buffers/historyLog.vim]: So - overall I think the line setting it globally should be removed, and this line should look like this:

```vim
let l:logLines = self.gitLines('log', '--max-count', get(g:, 'merginal_logCommitCount, -1), '--format=%h %aN%n%ci%n%s%n', self.branch)
```

**Comment by sfahlberg** [comment]
@idanarye thanks for all the comments- I'm pretty busy right now. I'll look into this more in late December

**Comment by sfahlberg** [comment]
@idanarye your comments make a lot of sense. I think it makes a lot more sense to use the getter method when you call `gitLines`. I updated the documentation (and fixed a typo). Let me know your thoughts on this.
